saleor/graphql/webhook/mutations/webhook_create.py

```diff
@@ -113,12 +98,38 @@ def clean_input(cls, info: ResolveInfo, instance, data, **kwargs):
                 "App doesn't exist or is disabled",
                 code=WebhookErrorCode.NOT_FOUND.value,
             )
-        clean_webhook_events(info, instance, cleaned_data)
+
+        subscription_query = None
         if query := cleaned_data.get("query"):
-            validate_query(query)
+            subscription_query = SubscriptionQuery(query)
+            if not subscription_query.is_valid:
+                raise_validation_error(
+                    field="query",
+                    message=subscription_query.error_msg,
+                    code=WebhookErrorCode.INVALID.value,
+                )
             instance.subscription_query = query
+
+        cls._clean_webhook_events(cleaned_data, subscription_query)
+
         return cleaned_data
 
+    @staticmethod
+    def _clean_webhook_events(data, subscription_query: Optional[SubscriptionQuery]):
+        # if `events` field is not empty, use this field. Otherwise get event types
+        # from `async_events` and `sync_events`. If the fields are also empty,
+        # parse events from `query`.
+        events = data.get("events", [])
+        if not events:
+            events += data.pop("async_events", [])
+            events += data.pop("sync_events", [])
+
+        if not events and subscription_query:
+            events = subscription_query.events
```

Comment:
Maybe if the subscription query is passed, we should ignore `events`, `asyncEvents`, and `syncEvents`? If the query is present, it should have precedence over other fields, I guess.

Reply:
Ignore this comment, when any of those "events" fields are passed, they should take precedence.